daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1672,7 +1672,7 @@ func (devices *DeviceSet) initDevmapper(doInit bool) (retErr error) {
 
 	// https://github.com/docker/docker/issues/4036
 	if supported := devicemapper.UdevSetSyncSupport(true); !supported {
-		if dockerversion.IAmStatic == "true" {
+		if linkmode.IsStatic {
```

Comment:
So, with my other comment in mind; I wonder if we actually _need_ the `linkmode` package; basically, we're passing a built-tag through a `const`; if this is now the only location we use it, and this is only to distinguish between "not supported because.. not supported" or "not supported because static".

Perhaps we should consider;

- changing `UdevSetSyncSupport` to return an error (instead of a `bool`), or to add a utility for that in `pkg/devicemapper`
- `pkg/devicemapper` is already using the `static_build` build-tag to implement different variations, so alternatively we could have some const there.
- Or just remove this code, and make the error message slightly less specific; the device mapper driver is already deprecated, so perhaps that's fine as well.

Reply:
- 👆 opened https://github.com/moby/moby/pull/44723 for this one